### PR TITLE
build: bump Go to v1.20

### LINKED ISSUE #2282: Bump go to 1.20
# Problem

Need to investigate first if a simple bump will result in any problems.

### PR DESCRIPTION
**Pull request by sainoe**
## Description

Closes: #2282

- bump golangci  to v1.52.2
- disable the "unused-parameter" warning from revive linter

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if API or client breaking change
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [x] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
* [x] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [x] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary 
* [x] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [x] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] confirmed `!` in the type prefix if API or client breaking change
* [x] confirmed all author checklist items have been addressed 
* [x] reviewed state machine logic
* [x] reviewed API design and naming
* [x] reviewed documentation is accurate
* [x] reviewed tests and test coverage

### CODE REVIEW

File: docs/getting-started/installation.md [hunk 1]
```diff
@@ -39,12 +39,13 @@ We suggest the following two ways to install Go. Check out the [official docs](h
 
 **Ubuntu:**
 
-At the time of this writing, the latest release is `1.18.10`. We're going to download the tarball, extract it to `/usr/local`, and export `GOROOT` to our `$PATH`
+At the time of this writing, the latest release is `1.2.10`. We're going to download the tarball, extract it to `/usr/local`, and export `GOROOT` to our `$PATH`
```
Comment (mmulji-ic) [inline docs/getting-started/installation.md]: version downloaded below is `1.20.3`, you mention `1.2.10` in the comment above, which is incorrect.

Comment (glnro) [inline docs/getting-started/installation.md]:
```suggestion
At the time of this writing, the latest release is `1.20.3`. We're going to download the tarball, extract it to `/usr/local`, and export `GOROOT` to our `$PATH`
```

**Review by glnro — APPROVED** [review]

**Review by glnro — DISMISSED** [review]
Forgot you need to change the makefile to update the version of go required to build.

**Comment by mpoke** [comment]
@sainoe could you please get the linter check pass?

**Comment by sainoe** [comment]
@mpoke I think that the linter passes. There are just some golangci warning msgs for some files changed by this PR.

**Comment by mmulji-ic** [comment]
@sainoe sonarcloud is failing [with](https://sonarcloud.io/project/security_hotspots?id=cosmos_gaia&pullRequest=2409&resolved=false&types=SECURITY_HOTSPOT&tab=code). I think that this is ok as this image is used only for testing. For other images we should use nonroot user.

Could you address the lint [issues](https://github.com/cosmos/gaia/actions/runs/4729757084/jobs/8392626018?pr=2409)

File: .goreleaser.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ env:
 
 before:
   hooks:
-    - go mod tidy -compat=1.18
+    - go mod tidy -compat=1.20
```
Comment (mmulji-ic) [inline .goreleaser.yml:9]: We're not using this at the moment, but still good to update.

**Review by mmulji-ic — APPROVED** [review]

**Review by MSalopek — APPROVED** [review]
Looks good!

Should any changes for `v9.0.x` should still use `go1.18`?

**Comment by mpoke** [comment]
> Should any changes for `v9.0.x` should still use `go1.18`?

Yes. Bumping go is state-machine breaking!

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -117,21 +117,25 @@ jobs:
...
-          go-version: 1.18
       - uses: technote-space/get-diff-action@v6.0.1
         with:
           PATTERNS: |
             **/**.go
             go.mod
             go.sum
+      - uses: actions/setup-go@v4
+        with:
+          go-version: 1.18
```
Comment (shaspitz) [inline .github/workflows/test.yml:128]: Why is this version 1.18?

**Pull request merged**